src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -254,10 +256,13 @@ def refreshCache(self):
 
         logging.info("Querying WMBS for jobs to be submitted...")
         logging.info("cachedJobIDs : %s" % len(self.cachedJobIDs))
-        if len(self.cachedJobIDs) < self.cacheRefreshSize:
+        if self.cacheRefreshSize == -1 or len(self.cachedJobIDs) < self.cacheRefreshSize or \
```

Comment:
Actually, if  it is set to 0, it will never get to be refreshed. since cache size has to go lower than the  cacheRefreshSize to be refreshed. :smile: